src/unit_tests/test_json_queue.c

```diff
@@ -0,0 +1,931 @@
+
+    /* success */
+
+    expect_string(__wrap_fopen, __filename, "/var/ossec/logs/alerts/alerts.json");
+    expect_string(__wrap_fopen, __modes, "r");
+    will_return(__wrap_fopen, fp);
+    will_return(__wrap_fseek, 1);
+    will_return(__wrap_fstat, 1);
+
+    int ret = jqueue_open(fileq, -1);
+
+    assert_int_equal(ret, 0);
+    assert_string_equal(fileq->file_name, "/var/ossec/logs/alerts/alerts.json");
+    assert_ptr_equal(fileq->fp, fp);
+}
+
+void test_jqueue_next_fail(void **state)
+{
+    struct aux_struct *aux = *state;
+
+    file_queue *fileq = aux->fileq;
+
+    /* jqueue_open fail */
+
+    expect_string(__wrap_fopen, __filename, "/var/ossec/logs/alerts/alerts.json");
+    expect_string(__wrap_fopen, __modes, "r");
+    will_return(__wrap_fopen, NULL);
+    expect_string(__wrap__merror, formatted_msg, "(1103): Could not open file '/var/ossec/logs/alerts/alerts.json' due to [(0)-(Success)].");
+
+    aux->json = jqueue_next(fileq);
+
+    assert_null(aux->json);
+    assert_string_equal(fileq->file_name, "/var/ossec/logs/alerts/alerts.json");
+    assert_null(fileq->fp);
+}
+
+void test_jqueue_next_success_newline(void **state)
+{
+    struct aux_struct *aux = *state;
+
+    file_queue *fileq = aux->fileq;
+    FILE *fp = aux->fp;
+
+    /* fgets success with \n */
+
+    expect_string(__wrap_fopen, __filename, "/var/ossec/logs/alerts/alerts.json");
+    expect_string(__wrap_fopen, __modes, "r");
+    will_return(__wrap_fopen, fp);
+    will_return(__wrap_fseek, 1);
+    will_return(__wrap_fstat, 1);
+
+    expect_value(__wrap_fgets, __n, OS_MAXSTR + 1);
+    expect_memory(__wrap_fgets, __stream, fp, sizeof(fp));
+    will_return(__wrap_fgets, "{\"Test\":\"Hello World 1\"}\n");
+    will_return(__wrap_fgets, "ok");
+
+    aux->json = jqueue_next(fileq);
+
+    assert_string_equal(cJSON_GetObjectItem(aux->json, "Test")->valuestring, "Hello World 1");
+    assert_string_equal(fileq->file_name, "/var/ossec/logs/alerts/alerts.json");
+    assert_ptr_equal(fileq->fp, fp);
+}
+
+void test_jqueue_next_success_no_newline(void **state)
+{
+    struct aux_struct *aux = *state;
+
+    file_queue *fileq = aux->fileq;
+    FILE *fp = aux->fp;
+
+    /* fgets success without \n */
+
+    expect_string(__wrap_fopen, __filename, "/var/ossec/logs/alerts/alerts.json");
+    expect_string(__wrap_fopen, __modes, "r");
+    will_return(__wrap_fopen, fp);
+    will_return(__wrap_fseek, 1);
+    will_return(__wrap_fstat, 1);
+
+    expect_value(__wrap_fgets, __n, OS_MAXSTR + 1);
+    expect_memory(__wrap_fgets, __stream, fp, sizeof(fp));
+    will_return(__wrap_fgets, "{\"Test\":\"Hello World 2\"}");
+    will_return(__wrap_fgets, "ok");
+
+    aux->json = jqueue_next(fileq);
+
+    assert_string_equal(cJSON_GetObjectItem(aux->json, "Test")->valuestring, "Hello World 2");
+    assert_string_equal(fileq->file_name, "/var/ossec/logs/alerts/alerts.json");
+    assert_ptr_equal(fileq->fp, fp);
+}
+
+void test_jqueue_close(void **state)
```

Comment:
Calling jqueue_open in this test is not necessary, we can simply pass a mock fileq with a random fp value and check it is NULL on return.